app/ui.js

```diff
@@ -936,12 +938,43 @@ const UI = {
         }
     },
 
+    readClipboard(callback) {
+      var readfuture = navigator.clipboard.readText();
+      readfuture
+        .then(text => callback(text))
+        .catch(() =>
+            console.error("Failed to read contents of clipboard (trying to copy to VNC clipboard)")
+        );
+    },
+
     clipboardReceive(e) {
         Log.Debug(">> UI.clipboardReceive: " + e.detail.text.substr(0, 40) + "...");
         document.getElementById('noVNC_clipboard_text').value = e.detail.text;
         Log.Debug("<< UI.clipboardReceive");
     },
 
+    enterVNC(e) {
+        UI.justEnteredVNC = true;
+    },
+
+    enterVncClick(e) {
+        UI.readClipboard(text => {
```

Comment:
Why not merge `readClipboard()` directly in to this function? It doesn't seem to be called elsewhere.